### PR TITLE
Add parallelism to publishing 

### PR DESCRIPTION
**Pull request by epananth**
### To double check:

* [ ] The right tests are in and and the right validation has happened.  Guidance:  https://github.com/dotnet/core-eng/tree/master/Documentation/Validation

Issue -> https://github.com/dotnet/arcade/issues/7314

Test 

  | Max 12 | Max 16 | Max 20 | Max 24
-- | -- | -- | -- | --
Runtime | https://dev.azure.com/dnceng/internal/_build/results?buildId=1112448   17mins 46 seconds | https://dnceng.visualstudio.com/internal/_build/results?buildId=1111250&view=logs&j=ba23343f-f710-5af9-782d-5bd26b102304&t=6e277ba4-1c1e-552d-b96f-db0aeb4be20a   17m 18 seconds | https://dnceng.visualstudio.com/internal/_build/results?buildId=1112563&view=logs&j=ba23343f-f710-5af9-782d-5bd26b102304&t=6e277ba4-1c1e-552d-b96f-db0aeb4be20a   13m 51 seconds | https://dnceng.visualstudio.com/internal/_build/results?buildId=1112670&view=logs&j=ba23343f-f710-5af9-782d-5bd26b102304&t=6e277ba4-1c1e-552d-b96f-db0aeb4be20a   15min 31 seconds

### CODE REVIEW

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/PublishArtifactsInManifestV3.cs [hunk 1]
```diff
@@ -63,6 +64,11 @@ public class PublishArtifactsInManifestV3 : PublishArtifactsInManifestBase
 
         public string SymbolsFeedOverride { get; set; }
 
+        /// <summary>
+        /// Maximum number of parallel uploads for the upload tasks
+        /// </summary>
+        public int MaximumClients { get; set; } = 20;
```
Comment (mmitche) [inline src/Microsoft.DotNet.Build.Tasks.Feed/src/PublishArtifactsInManifestV3.cs:70]: There's already a MaxClients in the base class, and I think we should refactor that so it's usable here and eliminate the other copy. I think doing the following makes sense:

- Change PublishArtifactsInManifestBase to be private specify MaxClients based on the UseStreamingPublishing switch:
  ```
        /// <summary>
        /// Maximum number of parallel uploads for the upload tasks.
        /// For streaming publishing, 20 is used as the most optimal.
        /// For non-streaming publishing, 16 is used (there are multiple sets of 16-parallel uploads)
        /// </summary>
        private int MaxClients { get { return UseStreamingPublishing ? StreamingPublishingMaxClients : NonStreamingPublishingMaxClients; } }
  ```
- Remove MaxClients property from PublishArtifactsInManifest - It's never set to its non-default value when calling the sdk publish task, and the value would mean different things based on the publishing type

Comment (epananth) [reply src/Microsoft.DotNet.Build.Tasks.Feed/src/PublishArtifactsInManifestV3.cs:70]: I think the maxClients should still be public, cos PublishArtifactsInManifestV3 still uses this.

Comment (mmitche) [reply src/Microsoft.DotNet.Build.Tasks.Feed/src/PublishArtifactsInManifestV3.cs:70]: Ahh yeah, correct. But make it a get-only property.

**Comment by epananth** [comment]
Test with new changes : https://dnceng.visualstudio.com/internal/_build/results?buildId=1114518&view=logs&j=ba23343f-f710-5af9-782d-5bd26b102304&t=6e277ba4-1c1e-552d-b96f-db0aeb4be20a

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/PublishArtifactsInManifest.cs [hunk 2]
```diff
@@ -94,7 +94,7 @@ public class PublishArtifactsInManifest : MSBuildTaskBase
         /// <summary>
         /// Maximum number of parallel uploads for the upload tasks
         /// </summary>
-        public int MaxClients { get; set; } = 16;
+        public int MaxClients { get; set; }
```
Comment (mmitche) [inline src/Microsoft.DotNet.Build.Tasks.Feed/src/PublishArtifactsInManifest.cs:97]: I think this can be removed altogether.

**Review by mmitche — APPROVED** [review]

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/PublishArtifactsInManifest.cs (hunk 2 above)
Comment (epananth) [reply src/Microsoft.DotNet.Build.Tasks.Feed/src/PublishArtifactsInManifest.cs:97]: Oops missed this one. Removing it

**Pull request merged**